docs/Tutorials/CCE.md

```diff
@@ -69,20 +69,104 @@ make CharacteristicExtract
+datasets with these exact names (including the `.dat` suffix):
+
+- `Beta.dat`
+- `DrJ.dat`
+- `DuR.dat`
+- `H.dat`
+- `J.dat`
+- `Q.dat`
+- `R.dat`
+- `U.dat`
+- `W.dat`
+
+Each dataset in the file must also have an attribute named `Legend` which
+is an ASCII-encoded null-terminated variable-length string. That is, the HDF5
+type is:
+
+```
+DATATYPE  H5T_STRING {
+  STRSIZE H5T_VARIABLE;
+  STRPAD H5T_STR_NULLTERM;
+  CSET H5T_CSET_ASCII;
+  CTYPE H5T_C_S1;
+}
+```
+
+This can be checked for a dataset by running
+
+```
+h5dump -a Beta.dat/Legend WorldtubeFile.h5
+```
+
+For the ordering of the data, we use spherical harmonic conventions documented
+by the ylm::Spherepack class. Each row must start with the time stamp, and the
+remaining values are the complex modes in m-varies-fastest format. For
+spin-weight zero Bondi variables (`Beta`, `R`, `DuR`, `W`), we omit the
+redundant negative-m modes and imaginary parts of the m=0 modes to save space on
+disk. Here is an example of a legend for the spin-weight zero variables:
+
+```
+"time", "Re(0,0)", "Re(1,0)", "Re(1,1)", "Im(1,1)", "Re(2,0)",
+"Re(2,1)", "Im(2,1)", "Re(2,2)", "Im(2,2)", "Re(3,0)", "Re(3,1)",
+"Im(3,1)", "Re(3,2)", "Im(3,2)", "Re(3,3)", "Im(3,3)", ...
+```
+
+For non-zero spin-weight Bondi variables (`J`, `DrJ`, `H`, `Q`, `U`) we must
+store all complex m-modes. Here is an example of a legend for variables where
+all complex m-modes must be specified:
+
+```
+"time", "Re(0,0)", "Im(0,0)", "Re(1,-1)", "Im(1,-1)", "Re(1,0)", "Im(1,0)",
+"Re(1,1)", "Im(1,1)", "Re(2,-2)", "Im(2,-2)", "Re(2,-1)", "Im(2,-1)", "Re(2,0)",
+"Im(2,0)", "Re(2,1)", "Im(2,1)", "Re(2,2)", "Im(2,2)", ...
+```
+
+We don't have strict requirement on the name of the H5 file that holds the
+worldtube data. However, it is recommended to name the H5 file `...CceRXXXX.h5`,
+where the `XXXX` is to be replaced by the zero-padded integer for which the
+extraction radius is equal to `XXXX`M. For instance, a 100M extraction should
+have filename `...CceR0100.h5`. If you do not adhere to this naming convention,
+you will need to specify the extraction radius in your YAML input file.
+
+\note This scheme of labeling files with the extraction radius is constructed
+for compatibility with worldtube data from the SXS Collaboration's SpEC code.
+
+### Converting to the required H5 format {#converting_worldtube_data}
+
+Unless you are using worldtube data that was generated from SpECTRE (or SpEC),
+it's possible that your worldtube data is not in the correct format. We allow
+conversion into our data format from a few other data formats using the
+[`ReduceCceWorldtube` executable provided](#acquiring_the_cce_module). These are
+
+- Nodal cartesian metric data (which we refer to as "metric nodal")
+- Modal cartesian metric data (which we refer to as "metric modal")
+- Nodal Bondi-Sachs data (which we refer to as "bondi nodal")
+
+Requirements for these data formats are listed below.
+
```

Comment:
We should probably state what we mean by `nodal` and `modal` because folks less familiar with spectral methods may not be familiar.